packages/xstate-vue/src/fsm.ts

```diff
@@ -23,5 +37,55 @@ export function useMachine<
     service.stop();
   });
 
-  return { state, send, service };
+  return { state, send: service.send, service };
+}
+
+const getServiceValue = <
+  TContext extends object,
+  TEvent extends EventObject = EventObject,
+  TState extends Typestate<TContext> = any
+>(
+  service: StateMachine.Service<TContext, TEvent, TState>
+): StateMachine.State<TContext, TEvent, TState> => {
+  let currentValue: StateMachine.State<TContext, TEvent, TState>;
+  service
+    .subscribe(state => {
+      currentValue = state;
+    })
+    .unsubscribe();
+  return currentValue!;
+};
+
+export function useService<
+  TContext extends object,
+  TEvent extends EventObject = EventObject,
+  TState extends Typestate<TContext> = any
+>(
+  service:
+    | StateMachine.Service<TContext, TEvent, TState>
+    | Ref<StateMachine.Service<TContext, TEvent, TState>>
+): {
+  state: Ref<StateMachine.State<TContext, TEvent, TState>>;
+  send: StateMachine.Service<TContext, TEvent, TState>['send'];
+  service: Ref<StateMachine.Service<TContext, TEvent, TState>>;
+} {
+  const serviceRef = isRef(service) ? service : ref(service);
+  const state = ref<StateMachine.State<TContext, TEvent, TState>>(
+    getServiceValue(serviceRef.value)
+  );
+
+  watch(serviceRef, (service, _, onCleanup) => {
+    state.value = getServiceValue(service);
+
+    const { unsubscribe } = service.subscribe(currentState => {
+      if (currentState.changed) {
+        state.value = currentState;
+      }
+    });
+    onCleanup(() => unsubscribe());
+  });
+
+  const send = (event: TEvent | TEvent['type']) => serviceRef.value.send(event);
+
+  return { state, send, service: serviceRef };
```

Comment:
it's a little bit weird that this return type is different from the one in `useMachine`, in here a `serviceRef` is returned and in `useMachine` it's just a `service`. It makes semantic sense, because `useMachine` can never change returned service, but this one can. It looks slightly inconsistent though, so maybe this one should return `serviceRef`? that would be inconsistent with `state` though (as it's ref as well) 

Changing the return type to a tuple would make the issue go away 😅

Reply:
Yeah, I also spent some time thinking about how to name it. The thing is, that with the composition API you are expected to get `Ref`s, when destructuring the object (e.g. `const { foo, bar } = useBaz()`) and using them in the vue `<template>` part will also automatically unwrap them.

So I think just leaving it as `service` is more beneficial than returning `serviceRef`.

By the way, the vue-apollo (which is maintained by a vue core member) project also does it like that in their [composition API implementation.](https://github.com/vuejs/vue-apollo/blob/v4/packages/vue-apollo-composable/src/useQuery.ts#L451-L453)